vassal-app/src/main/java/VASSAL/build/module/Map.java

```diff
@@ -2226,10 +2227,43 @@ public void centerAt(Point p, int dx, int dy) {
   /** Ensure that the given region (in map coordinates) is visible */
   public void ensureVisible(Rectangle r) {
     if (scroll != null) {
+      boolean bTriggerRecenter = false;
       final Point p = mapToComponent(r.getLocation());
-      r = new Rectangle(p.x, p.y,
-            (int) (getZoom() * r.width), (int) (getZoom() * r.height));
-      theMap.scrollRectToVisible(r);
+      final Rectangle rCurrent = theMap.getVisibleRect();
+      Rectangle rNorecenter = new Rectangle(0, 0);
+
+      // If r is already visible decide if unit is close enough to
+      // border to justify a recenter
+
+      // Close enough means a strip of the window along the edges whose
+      // width is a % of the edge to center of the window
+
+      // The % is defined in GlobalOptions.CENTER_ON_MOVE_SENSITIVITY
+      final double noRecenterPct = (100 - GlobalOptions.getInstance().centerOnOpponentsMoveSensitivity()) / 100;
+
+      // if r is within a band of  n%width/height of border, trigger recenter
+      rNorecenter.width = (int) round(rCurrent.width * noRecenterPct);
+      rNorecenter.height = (int) round(rCurrent.height * noRecenterPct);
+      rNorecenter.x = rCurrent.x + (int) round(rCurrent.width - rNorecenter.width) / 2;
+      rNorecenter.y = rCurrent.y + (int) round(rCurrent.height - rNorecenter.height) / 2;
+
+      if (p.x < rNorecenter.x) bTriggerRecenter = true;
+      if (p.x > (rNorecenter.x + rNorecenter.width)) bTriggerRecenter = true;
+      if (p.y < rNorecenter.y) bTriggerRecenter = true;
+      if (p.y > (rNorecenter.y + rNorecenter.height)) bTriggerRecenter = true;
+
```

Comment:
This unnecessarily evaluates conditions after you've already set `bTriggerRecenter` to true.

Better would be:
```
bTriggerRecenter = p.x < rNorecenter.x || p.x > (rNorecenter.x + rNorecenter.width) ||
                   p.y < rNorecenter.y || p.y > (rNorecenter.y + rNorecenter.height);
```